lumicks/pylake/scan.py

```diff
@@ -91,10 +91,137 @@ def _plot(self, image, frame=1, **kwargs):
             aspect=(image.shape[0] / image.shape[1]) * (x_um / y_um)
         )
 
-        plt.imshow(image, **{**default_kwargs, **kwargs})
+        if not image_handle:
+            image_handle = plt.imshow(image, **{**default_kwargs, **kwargs})
+        else:
+            # Updating the image data in an existing plot is a lot faster than re-plotting with `imshow`.
+            image_handle.set_data(image)
+
         plt.xlabel(f"{axis_label[axes[0]['axis']]} ($\\mu m$)")
         plt.ylabel(f"{axis_label[axes[1]['axis']]} ($\\mu m$)")
         if self.num_frames == 1:
             plt.title(self.name)
         else:
             plt.title(f"{self.name} [frame {frame}/{self.num_frames}]")
+
+        return image_handle
+
+    def _export_movie(self, plot_type, file_name, start_frame, end_frame, fps, **kwargs):
+        """Export a movie of a particular scan plot
+
+        Parameters
+        ----------
+        plot_type : str
+            One of the plot types "red", "green", "blue" or "rgb".
+        file_name : str
+            File name to export to.
+        start_frame : int
+            First frame in exported movie.
+        end_frame : int
+            Last frame in exported movie.
+        fps : int
+            Frame rate.
+        **kwargs
+            Forwarded to :func:`matplotlib.pyplot.imshow`."""
+        from matplotlib import animation, rcParams
+        import matplotlib.pyplot as plt
+
+        metadata = dict(title=self.name)
+        if "ffmpeg" in animation.writers:
+            writer = animation.writers['ffmpeg'](fps=fps, metadata=metadata)
+        elif "pillow" in animation.writers:
+            writer = animation.writers['pillow'](fps=fps, metadata=metadata)
+        else:
+            raise RuntimeError("You need either ffmpeg or pillow installed to export movies.")
+
+        start_frame = start_frame if start_frame else 1
+        end_frame = end_frame if end_frame else self.num_frames
+        end_frame += 1
```

Comment:
Why unconditionally add 1? Shouldn't it only be for the default value? `self.num_frames + 1`

Reply:
Good catch! Fixing!